R/fileupload.R

```diff
@@ -20,6 +20,104 @@
 # form upload, i.e. traditional HTTP POST-based file upload) doesn't work with
 # the websockets package's HTTP server at the moment.
 
+# The following file names are illegal on Windows. It's also bad to follow them
+# with an extension, such as NUL.txt
+# https://msdn.microsoft.com/en-us/library/aa365247.aspx
+# https://blogs.msdn.microsoft.com/oldnewthing/20031022-00/?p=42073
+illegalWindowsNames <-
+  c("AUX",
+    paste0("COM", 1:9),
+    "CON",
+    paste0("LPT", 1:9),
+    "NUL",
+    "PRN")
+
+# @details Removes strings in file names that could cause problems on Windows.
+# @param str A character vector to remove illegal strings from.
+# @param illegal The vector of words, by themselves or with an extension, to
+#   remove from \code{str}
+# @return \code{str} with any illegal words removed.
+removeIllegalWindowsFilenames <- function(str, illegal) {
```

Comment:
Filenames on Windows are case-insensitive; seems like this function should match case-insensitively as well.